lib/bolt/pal.rb

```diff
@@ -615,5 +615,50 @@ def run_plan(plan_name, params, executor = nil, inventory = nil, pdb_client = ni
     rescue Bolt::Error => e
       Bolt::PlanResult.new(e, 'failure')
     end
+
+    def lookup(key, targets, inventory, executor, _concurrency)
+      # Install the puppet-agent package and collect facts. Facts are
+      # automatically added to the targets.
+      in_plan_compiler(executor, inventory, nil) do |compiler|
+        compiler.call_function('apply_prep', targets)
+      end
+
+      overrides = {
+        bolt_inventory: inventory,
+        bolt_project:   @project
+      }
+
+      # Do a lookup with a catalog compiler, which uses the 'hierarchy' key in
+      # Hiera config.
+      results = targets.map do |target|
```

Comment:
This seems fine since the use case we're solving for is looking up a value on a limited set of targets for debugging, but we may want to parallelize this if people are using it to lookup lots of values.

Reply:
I think we can save that for future work when it's requested by a user, especially since I think it's unlikely users will be running this command frequently with large lists of targets that would impact performance.